app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts

```diff
@@ -78,14 +99,30 @@ export class WebhookStage {
       this.stage.customHeaders[customHeader.key] = customHeader.value;
     });
   }
+
+  public displayField(field: string): boolean {
+    return !this.preconfiguredProperties || !this.preconfiguredProperties.some(property => property === field);
+  }
+
+  public noUserConfigurableFields(): boolean {
+    if (this.preconfiguredProperties) {
+      if (this.viewState.waitForCompletion) {
+        return this.preconfiguredProperties.length >= 12;
```

Comment:
worth adding a comment here as to where that `12` comes from, especially if there's a chance additional properties might get added/removed in the future and we need to tweak this to keep it in sync.